### ISSUE TITLE
Warning when using a TFAutoModelWithLMHead logit outputs in the graph mode

### ISSUE THREAD

**Issue by mitramir55** [tensorflow/tensorflow#48451]
I've built a translation model following [this TensorFlow tutorial](https://www.tensorflow.org/tutorials/text/nmt_with_attention). The overall structure of my code is the same, although I added GPT in the decoder function to get help from this language model and create better translations. Right now this is how I import the model:
```
gpt_model = TFAutoModelWithLMHead.from_pretrained(model_name_or_path,
                                                  output_attentions=False,
                                                  return_dict=False,
                                                  output_hidden_states=False,
                                                  use_cache=False
                                                  )

gpt_tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)
```

and in the dencoder function, I give one input_id to the GPT model, and output the logits in the graph mode:

`gpt_model(input_id)[0][:, -1, :]`

 Right now my question is why do I get the following warning and how can I silence them properly when training in graph mode?
```
epoch  0
WARNING:tensorflow:The parameters `output_attentions`, `output_hidden_states` and `use_cache` cannot be updated when calling a model.They have to be set to True/False in the config object (i.e.: `config=XConfig.from_pretrained('name', output_attentions=True)`).
WARNING:tensorflow:AutoGraph could not transform <bound method Socket.send of <zmq.sugar.socket.Socket object at 0x7f90975dfde0>> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: module, class, method, function, traceback, frame, or code object was expected, got cython_function_or_method
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
```

and the following error:
```
WARNING:tensorflow:The parameters `output_attentions`, `output_hidden_states` and `use_cache` cannot be updated when calling a model.They have to be set to True/False in the config object (i.e.: `config=XConfig.from_pretrained('name', output_attentions=True)`).
WARNING:tensorflow:The parameter `return_dict` cannot be set in graph mode and will always be set to `True`.
```

The second warning doesn't seem consistent as I have already set all these values to False in the model config. I also tried decorating the model in the following way and then using it, but it didn't work either. (I don't know if it's even correct to do so)

`gpt_model = tf.autograph.experimental.do_not_convert(gpt_model)
`

I'm running all the code in Colab with GPU mode on.

Thanks in advance

**Comment by UsharaniPagadala** [comment]
@mitramir55 

Could you please fill the issue template

In order to expedite the trouble-shooting process, could you please provide the following details

OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
TensorFlow installed from (source or binary):
TensorFlow version:
Python version:
Installed using virtualenv? pip? conda?:
Bazel version (if compiling from source):
GCC/Compiler version (if compiling from source):
CUDA/cuDNN version:
GPU model and memory

and the exact sequence of commands / steps that you executed before running into the problem. Thanks!

**Comment by mitramir55** [comment]
Hi @UsharaniPagadala, thank you for your response.

I'm working with Google Colab and this problem occurs in all modes (with or without GPU or TPU) with the usual memory it offers to normal users (13 gigabytes).
I'm using:
TensorFlow 2.4.1
Python 3.7.10
In short, I run the GPT model (Persian GPT model) in the graph mode and generate logits for predicting the next tokens based on the target tokens (till time step t) in the target sentence. Then these logits are linearly combined with the seq2seq model output and together they predict the next token id.
This warning does not occur when I just run the seq2seq translation model and only appears when I generate tokens using GPT too.

[Here is a copy of the model and preprocessing steps](https://colab.research.google.com/drive/16dddt-NS4xmeYCCChhkgTvgnEf7ePZ3o?usp=sharing). I've titled every part of the code and you can go straight to the last section called "The Training". Please tell me if there is any more information needed. 


Thanks in advance.

**Comment by Saduf2019** [comment]
@mitramir55 
These two messages are just a warning, you can ignore them if you are not concerned. Basically, these messages will always be displayed everytime the graph node is executed, and only in graph mode.

**Comment by mitramir55** [comment]
Thank you for your response. I guess the warning prompt should be changed because each time I run my code it tells me to contact the TensorFlow team.
`
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, export AUTOGRAPH_VERBOSITY=10) and attach the full output.`

**Comment by Saduf2019** [comment]
@mitramir55 
Can you confirm if you are using:
```imported keras from tensorflow (from tensorflow import keras)```
Also, please take a look at [this comment](https://github.com/tensorflow/tensorflow/issues/45354#issuecomment-740039243) from a similar issue and check if it helps. [#47898] Thanks!

**Comment by mitramir55** [comment]
Yes, for all layers, loss, and the optimizer I use tf.keras.layers\optimizers\...
Thanks for sharing these links, but I was looking for a solution or a reason behind this type of warning instead of just suppressing it. Something like this is also printed out when training a simple text classification neural net in Colab (without the graph mode on):

![image](https://user-images.githubusercontent.com/53291220/117398023-a22c0c80-af12-11eb-8145-9ee18998016c.png)

**Comment by ymodak** [comment]
Apologies for the delay in resolving your issue here. I am afraid that the code snippet provided are insufficient to reproduce the problem and also the warning messages you are referring to arise from [transformers module](https://github.com/huggingface/transformers/issues).
https://github.com/huggingface/transformers/blob/a26f4d620874b32d898a5b712006a4c856d07de1/src/transformers/modeling_tf_utils.py#L295
>WARNING:tensorflow:The parameters `output_attentions`, `output_hidden_states` and `use_cache` cannot be updated when calling a model.They have to be set to True/False in the config object (i.e.: `config=XConfig.from_pretrained('name', output_attentions=True)`).
WARNING:tensorflow:The parameter `return_dict` cannot be set in graph mode and will always be set to `True`.

[Transformers discussion forum](https://discuss.huggingface.co/) and their github repo is suitable place for this question.
